conda/core/portability.py

```diff
@@ -92,7 +101,7 @@ def _update_prefix(original_data):
 
 
 def replace_prefix(
-    mode: FileMode, data: bytes, placeholder: str, new_prefix: str
+    mode: FileMode, data: bytes, placeholder: str, new_prefix: str, subdir: str
```

Comment:
This is a public function, so adding a new positional argument may break it's usage outside of the conda repository. Even though this probably isn't happening, we still need to be careful. I think you could make this default to `noarch` without changing the original behavior of the function being used without this keyword argument.